### PR TITLE
AV-144921 LetsEncrypt Certificate Management Profile with DNS-01 Challenge

### PR DESCRIPTION
**Pull request by pkoshtavmware**
AV-144921 LetsEncrypt Certificate Management Profile with DNS-01 Challenge

### CODE REVIEW

File: cert_mgmt/letsencrypt_mgmt_profile_with_dns.py [hunk 1]
```diff
@@ -0,0 +1,464 @@
+'''
+###
+# Name: letsencrypt_mgmt_profile.py
```
Comment (nikhilky) [inline cert_mgmt/letsencrypt_mgmt_profile_with_dns.py:3]: It should be letsencrypt_mgmt_profile_with_dns.py

File: cert_mgmt/letsencrypt_mgmt_profile_with_dns.py [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+# Version: 0.1.1
+# License: MIT
+#
+# Description -
+#     This is a python script used for automatically requesting and renewing certificates
+#     from and via Let's Encrypt.
```
Comment (nikhilky) [inline cert_mgmt/letsencrypt_mgmt_profile_with_dns.py:9]: mention dns-01 challenge and about the add_dns_record function.

File: cert_mgmt/letsencrypt_mgmt_profile_with_dns.py [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+
+            while(response["ChangeInfo"]["Status"]!='INSYNC'):
+                response = client.get_change(id=response["ChangeInfo"]["Id"])
+
+            print("Added dns text record")
+        except Exception as e:
+            raise Exception("Error adding dns txt record to vs {}",e)
+
+    def remove_dns_text_record(key_digest_64, txt_record_name):
+        client = boto3.client('route53', aws_access_key_id='XXXX',
```
Comment (nikhilky) [inline cert_mgmt/letsencrypt_mgmt_profile_with_dns.py:190]: Maybe add key_id and access_key as sensitive parameters in the script, and pass the whole kwargs to add/ remove_dns_text_record function.

**Review by nikhilky — CHANGES_REQUESTED** [review]

File: cert_mgmt/letsencrypt_mgmt_profile_with_dns.py (hunk 1 above)
Comment (pkoshtavmware) [reply cert_mgmt/letsencrypt_mgmt_profile_with_dns.py]: Done

File: cert_mgmt/letsencrypt_mgmt_profile_with_dns.py (hunk 2 above)
Comment (pkoshtavmware) [reply cert_mgmt/letsencrypt_mgmt_profile_with_dns.py]: Done

File: cert_mgmt/letsencrypt_mgmt_profile_with_dns.py (hunk 3 above)
Comment (pkoshtavmware) [reply cert_mgmt/letsencrypt_mgmt_profile_with_dns.py]: Its a nice suggestion. Done

**Review by nikhilky — APPROVED** [review]
LGTM

**Pull request merged**